### PR TITLE
Move DmFlags definition into separate module

### PR DESCRIPTION
**Pull request by mulkieran**

### CODE REVIEW

**Comment by mulkieran** [comment]
clippy failure.

**Comment by mulkieran** [comment]
@agrover review, please?

**Comment by mulkieran** [comment]
clippy should pass this time.

**Comment by mulkieran** [comment]
And it does! @agrover review, please?

File: src/dm.rs [hunk 1]
```diff
@@ -6,8 +6,6 @@
 // Their scope is the whole file because the bitflags! macro generates a
 // lot of stuff, and so placing the annotation right on the use of the
 // bitflags! macro does not actually cover the suspicious code generated.
-#![allow(redundant_field_names)]
```
Comment (agrover) [inline src/dm.rs]: the preceding comment is copied in the new file, can it be removed from here?

File: src/dm_flags.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// These errors originate in bitflags! macro and are not ours to fix.
```
Comment (agrover) [inline src/dm_flags.rs]: Are these clippy errors or regular rustc errors? Just curious. Might be nice to say.

Comment (agrover) [reply src/dm_flags.rs]: Still an issue with most recent bitflags crate?

Comment (mulkieran) [reply src/dm_flags.rs]: The most recent bitflags crate hasn't changed since these clippy errors were added.

Comment (mulkieran) [reply src/dm_flags.rs]: These must be clippy errors, because we use clippy annotations to ignore them.

File: src/dm.rs (hunk 1 above)
Comment (mulkieran) [reply src/dm.rs]: Yes.

**Comment by mulkieran** [comment]
@agrover review? Turns out clippy change means those two lints aren't triggered by bitflags! generated code, so the whole clippy thing can go away.

**Pull request merged**